src/diagnostics/diagnostic.jl

```diff
@@ -152,8 +152,14 @@ function add_diagnostic_variable!(;
     comments = "",
     compute!,
 )
-    haskey(ALL_DIAGNOSTICS, short_name) &&
-        error("diagnostic $short_name already defined")
+    haskey(ALL_DIAGNOSTICS, short_name) && @warn(
+        "overwriting diagnostic `$short_name` entry containing fields\n" *
+        "$(map(
+            fieldname -> "$(getfield(ALL_DIAGNOSTICS[short_name], fieldname))",
+            fieldnames(DiagnosticVariable)[1:(end - 1)],
```

Comment:
I'd say that we can leave the function (so that we remove the [1:(end-1)]. It will probably print the function name, which is not too bad.

Reply:
I changed it to use filter to exclude the function, and now the output looks like this from the coupler:
```
┌ Warning: overwriting diagnostic `mse` entry containing fields
│ ("mse", "Moist static energy", "moist_static_energy", "J/kg", "Moist static energy")
└ @ ClimaAtmos.Diagnostics ~/.julia/packages/ClimaAtmos/DeXjA/src/diagnostics/diagnostic.jl:155
```